docs/basics/part-7-using-docker-with-earthly.md

```diff
@@ -0,0 +1,541 @@
+    END
+
+```
+When we use the `--compose` flag, Earthly will start up the services defined in the `docker-compose` file for us. In this case, we built a separate image that copies in our test files and uses the command to run the tests as its `ENTRYPOINT`. We can then load this image into our `WITH DOCKER` command. Note that loading an image will not run it by default, we need to explicitly run the image after we load it.
+
+## More Examples
+
+<details open>
+<summary>JS</summary>
+
+To copy the files for [this example ( Part 7 )](https://github.com/earthly/earthly/tree/main/examples/tutorial/js/part7) run
+
+```bash
+earthly --artifact github.com/earthly/earthly/examples/tutorial/js:main+part7/part7 ./part7
+```
+In this example, we use `WITH DOCKER` to run a frontend app and backend api together using Earthly.
+
+The App
+
+`./app/package.json`
+
+```json
+{
+  "name": "example-js",
+  "version": "0.0.1",
+  "description": "Hello world",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "MPL-2.0",
+  "devDependencies": {
+    "webpack": "^4.42.1",
+    "webpack-cli": "^3.3.11"
+  },
+  "dependencies": {
+    "http-server": "^0.12.1"
+  }
+}
+```
+
+`./app/package-lock.json` (empty)
+
+```json
+```
+
+The code of the app might look like this
+
+`./app/src/index.js`
+
+```js
+async function getUsers() {
+
+    const response = await fetch('http://localhost:3080/api/users');
+    return await response.json();
+}
+
+function component() {
+  const element = document.createElement('div');
+  getUsers()
+    .then( users => {
+      element.innerHTML = `hello world <b>${users[0].first_name} ${users[0].last_name}</b>`
+    })
+
+	return element;
+}
+
+document.body.appendChild(component());
+```
+
+`./app/src/index.html`
+
+```html
+<!doctype html>
+<html>
+
+<head>
+    <title>Getting Started</title>
+</head>
+
+<body>
+    <script src="./main.js"></script>
+</body>
+
+</html>
+```
+And our api.
+
+`./api/package.json`
+
+```json
+{
+  "name": "api",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "express": "^4.17.1",
+    "http-proxy-middleware": "^1.0.4",
+    "pg": "^8.7.3"
+  }
+}
+```
+`./api/package-lock.json` (empty)
+
+```json
+```
+`./api/server.js`
+
+```js
+const express = require('express');
+const path = require('path');
+const cors = require("cors");
+const app = express(),
+bodyParser = require("body-parser");
+port = 3080;
+
+app.use(bodyParser.json());
+app.use(express.static(path.join(__dirname, '../my-app/build')));
+
+app.use(cors());
+
+const users = [
+  {
+    'first_name': 'Lee',
+    'last_name' : 'Earth'
+  }
+]
+
+app.get('/api/users', (req, res) => {
+  console.log('api/users called!')
+  res.json(users);
+});
+
+app.listen(port, 'localhost', () => {
```

Comment:
In containerized applications, usually, it's not `localhost`, but rather `0.0.0.0` instead. Otherwise nothing from outside the container would be able to access it.